Zotero/Scenes/Detail/CopyBibliography/ViewModels/CopyBibliographyActionHandler.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  CopyBibliographyActionHandler.swift
+//  Zotero
+//
+//  Created by Miltiadis Vasilakis on 27/12/23.
+//  Copyright © 2023 Corporation for Digital Scholarship. All rights reserved.
+//
+
+import UIKit
+import WebKit
+
+import CocoaLumberjackSwift
+import RxSwift
+
+struct CopyBibliographyActionHandler: ViewModelActionHandler {
+    typealias Action = CopyBibliographyAction
+    typealias State = CopyBibliographyState
+
+    private unowned let citationController: CitationController
+    private let disposeBag = DisposeBag()
+
+    init(citationController: CitationController) {
+        self.citationController = citationController
+    }
+
+    func process(action: CopyBibliographyAction, in viewModel: ViewModel<CopyBibliographyActionHandler>) {
+        switch action {
+        case .preload(let webView):
+            preload(webView: webView, in: viewModel)
+
+        case .cleanup:
+            citationController.finishCitation()
+        }
+
+        func preload(webView: WKWebView, in viewModel: ViewModel<CopyBibliographyActionHandler>) {
+            update(viewModel: viewModel) { state in
+                state.processingBibliography = true
+            }
+
+            let itemIds = viewModel.state.itemIds
+            let libraryId = viewModel.state.libraryId
+            let styleId = viewModel.state.styleId
+            let localeId = viewModel.state.localeId
+            let exportAsHtml = viewModel.state.exportAsHtml
+            citationController.prepare(webView: webView, for: itemIds, libraryId: libraryId, styleId: styleId, localeId: localeId)
+                .flatMap { [weak webView] _ -> Single<String> in
+                    guard let webView else { return Single.error(CitationController.Error.prepareNotCalled) }
+                    return citationController.bibliography(for: itemIds, format: .html, in: webView)
+                }
+                .flatMap { [weak webView] html -> Single<(String, String?)> in
+                    if exportAsHtml { return Single.just((html, nil)) }
+                    guard let webView else { return Single.error(CitationController.Error.prepareNotCalled) }
+                    return citationController.bibliography(for: itemIds, format: .text, in: webView).flatMap({ Single.just((html, $0)) })
+                }
+                .subscribe(onSuccess: { [weak viewModel] data in
```

Comment:
Using the convenience method, in case the object is not retained anymore, not even the part before the guard will be executed though. Of course in this instance it seems an unlikely edge case. Check https://github.com/zotero/zotero-ios/pull/827/commits/6230e70ee55e6670556eae5516d2c25628df4f0d